doc/api/domain.markdown

```diff
@@ -2,304 +2,265 @@
-
-```javascript
-// Much better!
-
-const cluster = require('cluster');
-const PORT = +process.env.PORT || 1337;
-
-if (cluster.isMaster) {
-  // In real life, you'd probably use more than just 2 workers,
-  // and perhaps not put the master and worker in the same file.
-  //
-  // You can also of course get a bit fancier about logging, and
-  // implement whatever custom logic you need to prevent DoS
-  // attacks and other bad behavior.
-  //
-  // See the options in the cluster documentation.
-  //
-  // The important thing is that the master does very little,
-  // increasing our resilience to unexpected errors.
-
-  cluster.fork();
-  cluster.fork();
-
-  cluster.on('disconnect', (worker) => {
-    console.error('disconnect!');
-    cluster.fork();
-  });
-
-} else {
-  // the worker
-  //
-  // This is where we put our bugs!
-
-  const domain = require('domain');
-
-  // See the cluster documentation for more details about using
-  // worker processes to serve requests.  How it works, caveats, etc.
-
-  const server = require('http').createServer((req, res) => {
-    var d = domain.create();
-    d.on('error', (er) => {
-      console.error('error', er.stack);
-
-      // Note: we're in dangerous territory!
-      // By definition, something unexpected occurred,
-      // which we probably didn't want.
-      // Anything can happen now!  Be very careful!
-
-      try {
-        // make sure we close down within 30 seconds
-        var killtimer = setTimeout(() => {
-          process.exit(1);
-        }, 30000);
-        // But don't keep the process open just for that!
-        killtimer.unref();
-
-        // stop taking new requests.
-        server.close();
-
-        // Let the master know we're dead.  This will trigger a
-        // 'disconnect' in the cluster master, and then it will fork
-        // a new worker.
-        cluster.worker.disconnect();
-
-        // try to send an error to the request that triggered the problem
-        res.statusCode = 500;
-        res.setHeader('content-type', 'text/plain');
-        res.end('Oops, there was a problem!\n');
-      } catch (er2) {
-        // oh well, not much we can do at this point.
-        console.error('Error sending 500!', er2.stack);
+The better approach is to handle thrown errors gracefully by sending an error
+response to the request that triggered the error, while letting other requests
+finish in their normal time.
+
+The `domain` module can work well with the `cluster` module. For instance,
+a master process can fork a new worker whenever a worker encounters an error.
+Use of the `domain` module allows this kind of response to occur in a more
+controlled manner.
+
+For instance, in the example `cluster` application below, worker processes
+are intentionally written to randomly throw errors. When these occur, the
+worker process shuts down immediately, leading to an ungraceful exit of the
+master process:
+
+    // An example of what not to do
```

Comment:
I'm not sure it's an example of what not to do. It's probably not what you want if you want your master process to be running when there's an error in one of the workers, but it's a valid pattern for other use cases.

This code is also not using domains, and is more a starting point to establish some context that the rest of the document builds upon to illustrate how to use domains.

I would suggest removing this mention, and instead write something like: 

```
Here's the first version of the sample code used in this section to illustrate patterns and anti-patterns when using domains [in the use case mentioned above]. One of the issues is that the master process exits with an error when any of the workers throws an error that is not caught. It can be a problem in some contexts, let's see how domains can help us build a more reliable program in the event of such errors.
```